b2/sync/folder.py

```diff
@@ -157,6 +164,10 @@ def _walk_relative_paths(self, local_dir, b2_dir, reporter):
                 )
             local_path = os.path.join(local_dir, name)
             b2_path = join_b2_path(b2_dir, name)
+
+            if filters_manager is not None and filters_manager.exclude(local_path):
```

Comment:
I think you need separate tests for files and directories.  A directory that matches an excludeFile regex should not be excluded, because that will also exclude all of the files under it.